Scripts/Presence_Absence_Analysis.R

```diff
@@ -27,49 +28,67 @@ sampling.df <- read.csv(file = sampling_path, header = TRUE)
 p.files <- list.files(path = presence, pattern = '*.json', full.names=TRUE)
 
 # turn into csv
-presence.df <- lapply(p.files, json_df, type = "_presence") %>% 
+presence.df <- lapply(p.files, json_df, type = "_presence") %>%
   dplyr::bind_rows() # collapses list of data frames into a single data frame
 
 #column names have a mix of "_" and ".", standardize to all being "_"
-names(presence.df) <- gsub(x = names(presence.df), 
-                           pattern = "\\.", 
-                           replacement = "_")  
+names(presence.df) <- gsub(x = names(presence.df),
+                           pattern = "\\.",
+                           replacement = "_")
 
 # write data frame to Results directory
 
-write.csv(presence.df, 
+write.csv(presence.df,
           file = presence_absence_matrix_path,
           row.names = FALSE) #no index
 
 #### merge with metadata ----
 #combine with metadata to get taxonomic hierarchy
 colnames(meta.df) #loaded in from paths.R
 
-presence.meta <- merge(presence.df, meta.df, 
-                       by.x = "file_name", 
-                       by.y = "original_file_name", 
+presence.meta <- merge(presence.df, meta.df,
+                       by.x = "file_name",
+                       by.y = "original_file_name",
                        all.x = TRUE, all.y = FALSE)
 
-#### 8. sampling after segmentation ----
+## remove images that don't have a scale ----
+
+sampling.df$Selection_Criteria[10] <- "Has ruler scale"
+
+unique(presence.meta$unit) #None means no scale was detected or extracted
+errors <- presence.meta[presence.meta$unit == "None",] %>% drop_na(unit)
```

Comment:
I had an error:
```
Error in drop_na(., unit) : could not find function "drop_na".
```
Here is one possible way to fix this issue:
```suggestion
errors <- presence.meta[presence.meta$unit == "None",] %>% tidyr::drop_na(unit)
```

Reply:
After making the above change I encountered another error on the same line:
```
Error in `chr_as_locations()`:
! Can't subset columns that don't exist.
✖ Column `unit` doesn't exist.
Backtrace:
     ▆
  1. ├─presence.meta[presence.meta$unit == "None", ] %>% ...

```
I think the above error is due to expecting there to be a `unit` column in the `segmentation/Morphology/Presence/*_presence.json` files like there is for the other metadata files (eg. `segmentation/Metadata/INHS_FISH_100201.json`).

Reply:
I rebuilt the morphology-analysis container with @thibaulttabarin's fixes from https://github.com/hdr-bgnn/Morphology-analysis/pull/39. I think the `unit` column now needs to be  renamed `ruler_unit` now.